lib/serializers/datastore.js

```diff
@@ -88,11 +88,30 @@ function fromDatastore(entity, options = {}) {
         Object.keys(entity).forEach((k) => {
             if (options.readAll || !{}.hasOwnProperty.call(schema.paths, k) || schema.paths[k].read !== false) {
                 let value = entity[k];
-                if ({}.hasOwnProperty.call(this.schema.paths, k)
-                    && this.schema.paths[k].type === 'datetime' && is.number(value)) {
+
+                if ({}.hasOwnProperty.call(this.schema.paths, k)) {
                     // During queries @google-cloud converts datetime to number
-                    value = new Date(value / 1000);
+                    if (this.schema.paths[k].type === 'datetime' && is.number(value)) {
+                        value = new Date(value / 1000);
+                    }
+
+                    // Sanitise embedded objects
+                    if (typeof this.schema.paths[k].excludeFromRead !== 'undefined'
+                        && is.array(this.schema.paths[k].excludeFromRead)
+                        && this.schema.paths[k].excludeFromRead.length > 0) {
```

Comment:
Not sure we need this last check as if its length is empty the `forEach` would not do anything.
But I think that what we need is to add `&& !options.readAll`. As, if we specify `readAll: true` when calling `plain()` we'd expect all the props to show up. Could you add it with a small test? thanks!

Reply:
Done.  I'm not sure why Github is not picking up the 2nd commit here.  Are you able to see it?